File: js/src/2639-findColumnWidth.js
/**
 * @param {number[][]} grid
 * @return {number[]}
 * 简单，遍历矩阵找到最大值
 */
// 60 ms, 在所有 JavaScript 提交中击败了92.22%
const findColumnWidth = function(grid) {
  const rowLen = grid.length;
  const columnLen = grid[0].length;
  const result = [];
  for (let i = 0; i < columnLen; i++) {
    let tmp = 0;
    for (let j = 0; j < rowLen; j++) {
      const curr = String(grid[j][i]).length;
      if (curr > tmp) {
        tmp = curr;
      }
    }
    result.push(tmp);
  }
  return result;
};

// console.log(findColumnWidth([[1], [22], [333]])); // [3]
// console.log(findColumnWidth([[-15, 1, 3], [15, 7, 12], [5, 6, -2]])); // [3,1,2]

export { findColumnWidth };

File: python/src/2639-findColumnWidth.py
from typing import List

class Solution:
    def findColumnWidth(self, grid: List[List[int]]) -> List[int]:
        result = []
        for i in range(0, len(grid[0])):
            tmp = 0
            for j in range(0, len(grid)):
                curr = len(str(grid[j][i]))
                if curr > tmp:
                    tmp = curr
            result.append(tmp)
        return result
